### PR TITLE
Prefill /yourschool data based on currently logged in user

### PR DESCRIPTION
**Pull request by drewsamnick**
This prefills to values of many fields on the /yourschool census form based on the currently logged in user's data. This only works if we make the page non-cacheable. Current traffic is not very high ("a few hundred visitors a day") but that could see big bursts depending on publicity.

### CODE REVIEW

**Comment by drewsamnick** [comment]
[Prefill school info if user if logged in](https://trello.com/c/AR1UY576/69-prefill-school-info-if-user-if-logged-in)

File: pegasus/sites.v3/code.org/public/yourschool.haml [hunk 1]
```diff
@@ -5,11 +5,25 @@ social:
...
   "og:description": "Expand computer science at your school or district. Join the thousands of schools who have already incorporated high quality computer science education into their curriculum and provide opportunities for the students in your local area."
 ---
+- require 'country_codes'
+- require 'state_abbr'
 %script{src: "https://cdn.optimizely.com/js/400912536.js"}
 - census_announcement = DCDO.get('census_announcement', nil)
 - if 'us' != hoc_detect_country
   - census_announcement ||= {}
   - census_announcement['hide_map'] = 'true'
+- if dashboard_user && dashboard_user[:school_info_id]
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/yourschool.haml]: If you get tired of seeing `-` you can also put all of this in a big `:ruby` block.

File: apps/src/sites/code.org/pages/public/yourschool.js [hunk 2]
```diff
@@ -12,13 +12,25 @@ registerReducers({isRtl, responsive});
...
         hideMap={$('#your-school').data("parameters-hide-map")}
+        userName={$('#your-school').data("parameters-user-name")}
+        userEmail={$('#your-school').data("parameters-user-email")}
+        isTeacher={$('#your-school').data("parameters-is-teacher")}
+        schoolCountry={$('#your-school').data("parameters-school-country")}
+        schoolId={rawSchoolId ? rawSchoolId.toString() : undefined}
+        schoolType={$('#your-school').data("parameters-school-type")}
+        schoolName={$('#your-school').data("parameters-school-name")}
+        schoolState={$('#your-school').data("parameters-school-state")}
+        schoolZip={rawSchoolZip ? rawSchoolZip.toString() : undefined}
```
Comment (joshlory) [inline apps/src/sites/code.org/pages/public/yourschool.js]: Worth organizing these into a structured object, along with `PropTypes.shape`?

File: apps/src/templates/census2017/YourSchool.jsx [hunk 3]
```diff
@@ -37,6 +37,15 @@ class YourSchool extends Component {
...
     alertUrl: PropTypes.string,
+    userName: PropTypes.string,
+    userEmail: PropTypes.string,
+    isTeacher: PropTypes.bool,
+    schoolCountry: PropTypes.string,
+    schoolId: PropTypes.string,
+    schoolType: PropTypes.string,
+    schoolName: PropTypes.string,
+    schoolState: PropTypes.string,
+    schoolZip: PropTypes.string,
```
Comment (joshlory) [inline apps/src/templates/census2017/YourSchool.jsx]: Can we share this prop validation with the list in `CensusForm`?

File: apps/src/templates/census2017/YourSchool.jsx [hunk 4]
```diff
@@ -94,7 +103,17 @@ class YourSchool extends Component {
...
+        <CensusForm
+          userName={this.props.userName}
+          userEmail={this.props.userEmail}
+          isTeacher={this.props.isTeacher}
+          schoolCountry={this.props.schoolCountry}
+          schoolId={this.props.schoolId}
+          schoolType={this.props.schoolType}
+          schoolName={this.props.schoolName}
+          schoolState={this.props.schoolState}
+          schoolZip={this.props.schoolZip}
```
Comment (joshlory) [inline apps/src/templates/census2017/YourSchool.jsx]: Organizing this under an object would mean only 1 or 2 things to pass in here.

File: pegasus/sites.v3/code.org/public/yourschool.haml [hunk 5]
```diff
@@ -5,11 +5,25 @@ social:
...
+  - census_announcement ||= {}
+  - census_announcement['user_name'] = dashboard_user[:name] if dashboard_user[:name]
+  - census_announcement['user_email'] = dashboard_user[:email] if dashboard_user[:email]
+  - census_announcement['is_teacher'] = 'true' if dashboard_user[:user_type] == 'teacher'
+  - school_info = DASHBOARD_DB[:school_infos].where(id: dashboard_user[:school_info_id]).first
```
Comment (joshlory) [inline pegasus/sites.v3/code.org/public/yourschool.haml]: Not a huge fan of `DASHBOARD_DB` calls inside Haml views.  I know we talked this through already, but prefer an AJAX call to a Dashboard API, so we're not reaching into the Dashboard DB from random Pegasus pages.

**Comment by drewsamnick** [comment]
We've decided to back off on loading the user data from the DB. Instead the code now allows those values to be passed in as parameters on the request. The idea here is that the only people who would benefit from pre-filling the data are code.org teachers and that mainly they would be directed to the /yourschool page via the link in the CensusTeacherBanner. We already have the school data at that point and can just put it into the link.

File: apps/src/sites/code.org/pages/public/yourschool.js (hunk 2 above)
Comment (drewsamnick) [reply apps/src/sites/code.org/pages/public/yourschool.js]: Done.

File: apps/src/templates/census2017/YourSchool.jsx (hunk 3 above)
Comment (drewsamnick) [reply apps/src/templates/census2017/YourSchool.jsx]: Done.

**Pull request merged**